infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/GitUserProfileProvisioner.java

```diff
@@ -0,0 +1,160 @@
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.user.server.PreferenceManager;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+
+@Singleton
+public class GitUserProfileProvisioner implements ConfigurationProvisioner<KubernetesEnvironment> {
+
+  private final String GIT_CONFIG_MAP_NAME_SUFFIX = "-gitconfig";
+
+  private static final String GIT_BASE_CONFIG_PATH = "/etc/";
+  private static final String GIT_CONFIG = "gitconfig";
+  private static final String GIT_CONFIG_PATH = GIT_BASE_CONFIG_PATH + GIT_CONFIG;
+  private static final String PREFERENCES_KEY_FILTER = "theia-user-preferences";
+  private static final String GIT_USER_NAME_PROPERTY = "git.user.name";
+  private static final String GIT_USER_EMAIL_PROPERTY = "git.user.email";
+  private static final String CONFIG_MAP_VOLUME_NAME = "gitconfigvolume";
+
+  private PreferenceManager preferenceManager;
+
+  @Inject
+  public GitUserProfileProvisioner(PreferenceManager preferenceManager) {
+    this.preferenceManager = preferenceManager;
+  }
+
+  @Override
+  public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+    getPreferenceValueOrNull(PREFERENCES_KEY_FILTER)
+        .ifPresent(
+            preferenceJsonValue -> {
+              Map<String, String> theiaPreferences = getMapFromJsonObject(preferenceJsonValue);
+
+              getGlobalGitConfigFileContent(
+                      theiaPreferences.get(GIT_USER_NAME_PROPERTY),
+                      theiaPreferences.get(GIT_USER_EMAIL_PROPERTY))
+                  .ifPresent(
+                      gitConfigFileContent -> {
+                        String gitConfigMapName =
+                            identity.getWorkspaceId() + GIT_CONFIG_MAP_NAME_SUFFIX;
+
+                        doProvisionGlobalGitConfig(gitConfigMapName, gitConfigFileContent, k8sEnv);
+                      });
+            });
+  }
+
+  private Map<String, String> getMapFromJsonObject(String json) {
+    Type stringMapType = new TypeToken<Map<String, String>>() {}.getType();
+
+    return new Gson().fromJson(json, stringMapType);
+  }
+
+  private Optional<String> getPreferenceValueOrNull(String keyFilter)
+      throws InfrastructureException {
+    try {
+      String userId = EnvironmentContext.getCurrent().getSubject().getUserId();
+      Map<String, String> preferencesMap = preferenceManager.find(userId, keyFilter);
+
+      return ofNullable(preferencesMap.get(PREFERENCES_KEY_FILTER));
```

Comment:
This function is a bit odd for me. It is a private function that is used with `PREFERENCES_KEY_FILTER` as keyFilter only. And in this like it refers to `PREFERENCES_KEY_FILTER` directly. I feel that either it should not refer to PREFERENCES_KEY_FILTER or not have a keyFilter.

Reply:
Yes, you're right! Thanks for the pointing. `keyFilter` argument actually has the same value as `PREFERENCES_KEY_FILTER ` has. Will replace constant value with argument value.

Reply:
Also will update name of the function, `getPreferenceValue` is more preferred name for it.